lib/lua/src/lstrlib.c

```diff
@@ -854,6 +854,12 @@ static int str_format (lua_State *L)
                 modp_itoa10((LUA_INTFRM_T)luaL_checknumber(L, arg),buff);
                 break;
             }
+            case 'x':
+            case 'X': {
+                addintlen(form);
+                modp_itoaX((LUA_INTFRM_T)luaL_checknumber(L, arg),buff,16);
```

Comment:
It doesn't look like the new function uitoa16 is used/needed?!

Reply:
You are correct. I made this change a long time ago and forgot that the lua code already had the function I needed. Thanks for finding that!